lib/elastic_apm/agent.rb

```diff
@@ -28,7 +28,8 @@ def self.start(config)
 
       LOCK.synchronize do
         return @instance if @instance
-        @instance = new(config).start
+
+        @instance = new(config.freeze).start
```

Comment:
You probably want to remove the `freeze` from https://github.com/elastic/apm-agent-ruby/blob/master/lib/elastic_apm/config.rb#L66 then.

Reply:
I didn't? wut.